File: Find Minimum in Rotated Sorted Array/Leetcode_153.cpp
class Solution {
public:
    int findMin(vector<int>& A) {
        int lo =0, hi = A.size()-1;
        while (hi>0 && A[lo]==A[hi]) hi--;
        int mid;
        while (lo<hi){
            mid = lo + (hi-lo)/2;
            if (A[mid] > A[hi]) lo=mid+1;
            else hi=mid;
        }
        return A[lo];
    }
};

File: Find Minimum in Rotated Sorted Array/Leetcode_153.py
class Solution:
    def findMin(self, A: List[int]) -> int:
        left, right = 0, len(A)-1
        while left < right:
            mid = (left+right)//2
            if A[mid] > A[right]:
                left = mid+1
            else:
                right = mid
        return A[left]
